doc/services/retention/index.rst

```diff
@@ -0,0 +1,199 @@
+.. _retention_api:
+
+Retention System
+################
+
+The retention system provides an API which allows applications to read/write
+data from/to memory areas or devices which retain the data whilst the device is
+powered. This is designed for sharing information between different
+applications or for a single application when a device reboots without losing
+state information, the contents of the data should not persist in the event of
+a power failure (or possibly with some low power modes on some devices) nor
+should it be stored to a non-volatile backing store such as :ref:`flash_api`,
+:ref:`eeprom_api` or battery-backed RAM.
+
+The retention system builds on top of the retained data driver, and adds
+additional software-level features to it for ensuring the validity of data.
+Optionally, a magic header can be used to check if the front of
+the retained data memory section contains this specific value, and an optional
+checksum (1, 2 or 4-bytes in size) of the stored data can be appended to the
+end of the data. Additionally, the retention system API allows partitioning of
+retained data sections into multiple distinct areas, for example a 64-byte
+retained data area could be split up into 4 bytes for a boot mode, 16 bytes
+for a timestamp, 44 bytes for a last log message - all of these sections can
+be accessed or updated independently. The prefix and checksum can be set
+per-instance via device tree configuration.
+
+Device tree setup
+*****************
+
+To use the retention system, a retained data driver must be setup for the board
+you are using, there is a zephyr driver which can be used which will use some
```

Comment:
Which zephyr driver? 

From my understanding of the code this must be `retained_mem_zephyr_ram.c` driver. Correct? If that is correct, please state this.

Reply:
No, any retained memory driver.